src/content-scope-features.js

```diff
@@ -5,30 +5,23 @@ import { PerformanceMonitor } from './performance'
 import injectedFeaturesCode from 'ddg:runtimeInjects'
 import platformFeatures from 'ddg:platformFeatures'
 
-function shouldRun () {
-    // don't inject into non-HTML documents (such as XML documents)
-    // but do inject into XHTML documents
-    // Should check HTMLDocument as Document is an alias for XMLDocument also.
-    if (document instanceof HTMLDocument === false && (
-        document instanceof XMLDocument === false ||
-        document.createElement('div') instanceof HTMLDivElement === false
-    )) {
-        return false
-    }
-    return true
-}
-
 let initArgs = null
 const updates = []
 const features = []
 const alwaysInitFeatures = new Set(['cookie'])
 const performanceMonitor = new PerformanceMonitor()
 
+// The document type is checked as soon as possible, then provided as a load
+// argument. That way, it minimises the risk of a website breaking our
+// instanceof checks by changing document.__proto__.
+let isHTMLDocument = null
```

Comment:
Why not just import the var here than wire it through args adding a bunch of complexity?

Reply:
Equally the simplest change is:
```suggestion
const shouldRun = shouldRun()
```

And just change the code to use the variable below:
`if (!shouldRun) {`

Reply:
(equally once tested we should keep your refactor of the method as it's way more readable I've just left it in how it was for so long)

Reply:
Done